aimmo-game/simulation/levels/decoders.py

```diff
@@ -0,0 +1,75 @@
+import abc
+
+from simulation.location import Location
+from simulation.world_map import WorldMapStaticSpawnDecorator
+from simulation.world_map import Cell
+
+from simulation.cell import *
+
+from simulation.pickups import HealthPickup
+from simulation.pickups import InvulnerabilityPickup
+from simulation.pickups import DamagePickup
+
+class Decoder():
+    """
+        See @JsonLevelGenerator and @Levels first.
+
+        A Decorer is a class that receives a Json formatted as in levels/models
+        and decodes it, altering the state of the world_map.
+
+        Decoders are used to translate a Level from the JSON format to the internal
+        state of a map(i.e. a WorldMap).
+
+        An example of a JSon format is:
+          {
+            "code": "1",
+            "id" : "5",
+            "x" : "3",
+            "y" : "3"
+          }
+    """
+    __metaclass__ = abc.ABCMeta
+
+    def __init__(self, code):
+        self.code = code
+
+    @abc.abstractmethod
+    def decode(self, json, world_map):
+        pass
+
+def get_sprite(json):
+    if "sprite" in json:
+        return json["sprite"]
```

Comment:
please use https://docs.python.org/2/library/stdtypes.html#dict.get to get the value here :)